### PR TITLE
Re run of results and initial write up.

### PR DESCRIPTION
**Pull request by drvinceknight**
Going to run the script to generate the cache :+1:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,34 @@
-# axelrod-moran
+# A numerical study of fixation probabilities for strategies in the Iterated Prisoner's Dilemma
 
-Requires Python 3 or an installation of [pathlib](https://github.com/mcmtroffaes/pathlib2)
+This is a paper giving an extensive numerical analysis of fixation probabilities
+for the Iterated Prisoner's Dilemma.
```
Comment (marcharper) [inline README.md]: of the Moran processes where the fitness landscape it determined by the IPD

File: environment.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
+name: moran
```
Comment (marcharper) [inline environment.yml:1]: do we need this file in source control?

File: src/generate_cache.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    outcomes = defaultdict(Counter)
+    for row in reader:
+        p1, p2, s1, s2, count = row
+        outcomes[(p1, p2)][(float(s1), float(s2))] = int(count)
+        outcomes[(p2, p1)][(float(s2), float(s1))] = int(count)
+    return outcomes
+
+def generate_matchups(players):
+    # Want the triangular product
+    for i in range(len(players)):
```
Comment (marcharper) [inline src/generate_cache.py]: use `generate_matchup_indices` here?

**Review by marcharper — APPROVED** [review]

File: environment.yml (hunk 2 above)
Comment (drvinceknight) [reply environment.yml:1]: I suggest we do, when we get the paper out there we'll basically be saying: if you use this exact environment (so for example the jupyter notebooks are happy with the env name etc) and run the code then you will get the same results we do.

**Pull request merged**